src/main/java/net/wurstclient/mixin/ClientPlayerEntityMixin.java

```diff
@@ -96,7 +97,11 @@ private boolean wrapHasForwardMovement(Input input,
 		ordinal = 0), method = "tickMovement()V")
 	private void onTickMovementItemUse(CallbackInfo ci)
 	{
-		if(WurstClient.INSTANCE.getHax().noSlowdownHack.isEnabled())
+		if(WurstClient.INSTANCE.getHax().noSlowdownHack.noItemSlowness()
+			|| (WurstClient.INSTANCE.getHax().noSlowdownHack
+				.noItemSlownessExceptShields()
+				&& ((ClientPlayerEntity)(Object)this).getActiveItem()
+					.getUseAction() != UseAction.BLOCK))

```

Comment:
@coderabbitai Good idea, bad example. Creating a local variable for `WurstClient.INSTANCE.getHax().noSlowdownHack` would improve readability in this case.

But also, it seems odd that there are two separate methods `noItemSlowness()` and `noItemSlownessExceptShields()` here that both need to be called together like that. A better solution might be to have a single `noItemSlowness(boolean shield)` method, which would handle the special case for shields internally.

Reply:
I was thinking more like this:

```java
// The cast to Object and back is necessary because this code is inside of a mixin.
boolean shield = ((ClientPlayerEntity)(Object)this).getActiveItem().getUseAction() == UseAction.BLOCK;
if(WurstClient.INSTANCE.getHax().noSlowdownHack.noItemSlowness(shield))
	hideNextItemUse = true;
```

Or, if keeping the two methods:

```java
NoSlowdownHack noSlowdown = WurstClient.INSTANCE.getHax().noSlowdownHack;
boolean shield = ((ClientPlayerEntity)(Object)this).getActiveItem().getUseAction() == UseAction.BLOCK;
if(noSlowdown.noItemSlowness() || (noSlowdown.noItemSlownessExceptShields() && shield)
	hideNextItemUse = true;
```

But the first solution is much cleaner.